remoteconfig/client.go

```diff
@@ -0,0 +1,223 @@
+package remoteconfig
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"firebase.google.com/go/v4/internal"
+)
+
+const (
+	defaultRemoteConfigEndpoint = "https://firebaseremoteconfig.googleapis.com/v1"
+)
+
+// Client .
+type Client struct {
+	hc        *internal.HTTPClient
+	projectID string
+}
+
+func (c *Client) getRootURL() string {
+	return fmt.Sprintf("%s/projects/%s/remoteConfig", defaultRemoteConfigEndpoint, c.projectID)
+}
+
+// GetRemoteConfig https://firebase.google.com/docs/reference/remote-config/rest/v1/projects/getRemoteConfig
+func (c *Client) GetRemoteConfig(versionNumber string) (*Response, error) {
+	var opts []internal.HTTPOption
+
+	// Optional. Version number of the RemoteConfig to look up.
+	// If not specified, the latest RemoteConfig will be returned.
+	if versionNumber != "" {
+		opts = append(opts, internal.WithQueryParam("versionNumber", versionNumber))
+	}
+
+	var data RemoteConfig
+
+	resp, err := c.hc.DoAndUnmarshal(
+		context.Background(),
+		&internal.Request{
+			Method: http.MethodGet,
+			URL:    c.getRootURL(),
+			Opts:   opts,
+		},
+		&data,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &Response{
+		RemoteConfig: data,
+		Etag:         resp.Header.Get("Etag"),
+	}
+
+	return result, nil
+}
+
+// UpdateRemoteConfig https://firebase.google.com/docs/reference/remote-config/rest/v1/projects/updateRemoteConfig
+func (c *Client) UpdateRemoteConfig(eTag string, validateOnly bool) (*Response, error) {
+	if eTag == "" {
+		eTag = "*"
+	}
+
+	var opts []internal.HTTPOption
+	opts = append(opts, internal.WithHeader("If-Match", eTag))
+	opts = append(opts, internal.WithQueryParam("validateOnly", strconv.FormatBool(validateOnly)))
+
+	var data RemoteConfig
+
+	resp, err := c.hc.DoAndUnmarshal(
+		context.Background(),
+		&internal.Request{
+			Method: http.MethodPut,
+			URL:    c.getRootURL(),
+			Opts:   opts,
+		},
+		&data,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &Response{
+		RemoteConfig: data,
+		Etag:         resp.Header.Get("Etag"),
+	}
+
+	return result, nil
+}
+
+// ListVersions https://firebase.google.com/docs/reference/remote-config/rest/v1/projects.remoteConfig/listVersions
+func (c *Client) ListVersions(options *ListVersionsOptions) (*ListVersionsResponse, error) {
```

Comment:
This should change as:
`func (c *Client) Versions(ctx context.Context, options *ListVersionsOptions) (*VersionIterator, error)`

Reply:
Reconsidered and this looks like a good first step. It is fine to leave this as it is. No need to make further changes. Please ignore my comment.